src/layers/data_type_layer.cpp

```diff
@@ -781,28 +812,24 @@ void data_type_layer<InputTensorDataType, OutputTensorDataType>::
 fp_setup_inputs(El::Int mini_batch_size) {
   if (get_num_parents() < 1) { return; }
 
-  // Determine distributed matrix alignment
-  const auto& alignment_dist = get_parent_layer().get_activations(*this).DistData();
```

Comment:
Alignment involves overwriting the matrix's `El::Grid`:

https://github.com/LLNL/Elemental/blob/3b7bfb8a8fa6182107fe22a248e1f19c9692692a/src/core/DistMatrix/ElementMatrix.cpp#L252

I believe the only case where matrices have non-default alignment is with the model-parallel slice layer. That case will still be correct with this PR, but it will pessimize by performing communication to restore default alignment.